CMakeLists.txt

```diff
@@ -55,39 +55,41 @@ set(THREADS_PREFER_PTHREAD_FLAG ON)
 find_package(Threads REQUIRED)
 ################################################################################
 # Extract cluon-msc from cluon-complete.hpp.
+
+execute_process(COMMAND wget -P ${CLUON_PATH} https://raw.githubusercontent.com/chrberger/libcluon/gh-pages/headeronly/${CLUON_COMPLETE})
+
 add_custom_command(OUTPUT ${CMAKE_BINARY_DIR}/cluon-msc
     WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
-    COMMAND ${CMAKE_COMMAND} -E create_symlink ${CMAKE_CURRENT_SOURCE_DIR}/src/${CLUON_COMPLETE} ${CMAKE_BINARY_DIR}/cluon-complete.hpp
+    COMMAND ${CMAKE_COMMAND} -E create_symlink ${CLUON_PATH}/${CLUON_COMPLETE} ${CMAKE_BINARY_DIR}/cluon-complete.hpp
     COMMAND ${CMAKE_COMMAND} -E create_symlink ${CMAKE_BINARY_DIR}/cluon-complete.hpp ${CMAKE_BINARY_DIR}/cluon-complete.cpp
     COMMAND ${CMAKE_CXX_COMPILER} -o ${CMAKE_BINARY_DIR}/cluon-msc ${CMAKE_BINARY_DIR}/cluon-complete.cpp -std=c++14 -pthread -D HAVE_CLUON_MSC
-    DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/src/${CLUON_COMPLETE})
+    DEPENDS ${CLUON_PATH}/${CLUON_COMPLETE})
 
 ################################################################################
 # Generate opendlv-standard-message-set.{hpp,cpp} from ${OPENDLV_STANDARD_MESSAGE_SET} file.
-add_custom_command(OUTPUT ${CMAKE_BINARY_DIR}/opendlv-standard-message-set.cpp
+add_custom_command(OUTPUT ${CMAKE_BINARY_DIR}/opendlv-standard-message-set.hpp
     WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
-    COMMAND ${CMAKE_BINARY_DIR}/cluon-msc --cpp-sources --cpp-add-include-file=opendlv-standard-message-set.hpp --out=${CMAKE_BINARY_DIR}/opendlv-standard-message-set.cpp ${CMAKE_CURRENT_SOURCE_DIR}/src/${OPENDLV_STANDARD_MESSAGE_SET}
-    COMMAND ${CMAKE_BINARY_DIR}/cluon-msc --cpp-headers --out=${CMAKE_BINARY_DIR}/opendlv-standard-message-set.hpp ${CMAKE_CURRENT_SOURCE_DIR}/src/${OPENDLV_STANDARD_MESSAGE_SET}
-    DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/src/${OPENDLV_STANDARD_MESSAGE_SET} ${CMAKE_BINARY_DIR}/cluon-msc)
+    COMMAND ${CMAKE_BINARY_DIR}/cluon-msc --cpp --out=${CMAKE_BINARY_DIR}/opendlv-standard-message-set.hpp ${CMAKE_CURRENT_SOURCE_DIR}/include/${OPENDLV_STANDARD_MESSAGE_SET}
+    DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/include/${OPENDLV_STANDARD_MESSAGE_SET} ${CMAKE_BINARY_DIR}/cluon-msc)
 
 # Add current build directory as include directory as it contains generated files.
 include_directories(SYSTEM ${CMAKE_BINARY_DIR})
 include_directories(SYSTEM ${CMAKE_CURRENT_SOURCE_DIR}/thirdparty)
```

Comment:
This will give you a warning/error when compiling since the directory no longer exists.
`[ 45%] Building CXX object CMakeFiles/cfsd18-perception-detectcone-core.dir/src/collector.cpp.o
cc1plus: error: /opt/sources/thirdparty: No such file or directory [-Werror=missing-include-dirs]
cc1plus: error: /opt/sources/thirdparty: No such file or directory [-Werror=missing-include-dirs]
cc1plus: error: /opt/sources/thirdparty: No such file or directory [-Werror=missing-include-dirs]
cc1plus: all warnings being treated as errors
make[2]: *** [CMakeFiles/cfsd18-perception-detectcone-core.dir/build.make:89: CMakeFiles/cfsd18-perception-detectcone-core.dir/src/cone.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
In file included from /usr/local/include/tiny_dnn/xtensor/xcontainer.hpp:17,
                 from /usr/local/include/tiny_dnn/xtensor/xarray.hpp:17,
                 from /usr/local/include/tiny_dnn/util/util.h:24,
                 from /usr/local/include/tiny_dnn/lossfunctions/loss_function.h:12,
                 from /usr/local/include/tiny_dnn/network.h:26,
                 from /usr/local/include/tiny_dnn/tiny_dnn.h:11,
                 from /opt/sources/include/detectcone.hpp:46,
                 from /opt/sources/src/detectcone.cpp:19:
/usr/local/include/tiny_dnn/xtensor/xiterable.hpp: In member function 'xt::xiterable<D>::layout_iterator<L> xt::xiterable<D>::storage_begin()':
/usr/local/include/tiny_dnn/xtensor/xiterable.hpp:740:19: error: expected primary-expression before ')' token
   return begin<L>();
                   ^
/usr/local/include/tiny_dnn/xtensor/xiterable.hpp: In member function 'xt::xiterable<D>::layout_iterator<L> xt::xiterable<D>::storage_end()':
/usr/local/include/tiny_dnn/xtensor/xiterable.hpp:746:17: error: expected primary-expression before ')' token
   return end<L>();
                 ^
/usr/local/include/tiny_dnn/xtensor/xiterable.hpp: In member function 'xt::xiterable<D>::reverse_layout_iterator<L> xt::xiterable<D>::storage_rbegin()':
/usr/local/include/tiny_dnn/xtensor/xiterable.hpp:753:20: error: expected primary-expression before ')' token
   return rbegin<L>();
                    ^
/usr/local/include/tiny_dnn/xtensor/xiterable.hpp: In member function 'xt::xiterable<D>::reverse_layout_iterator<L> xt::xiterable<D>::storage_rend()':
/usr/local/include/tiny_dnn/xtensor/xiterable.hpp:760:18: error: expected primary-expression before ')' token
   return rend<L>();
                  ^
In file included from /usr/local/include/tiny_dnn/xtensor/xcontainer.hpp:17,
                 from /usr/local/include/tiny_dnn/xtensor/xarray.hpp:17,
                 from /usr/local/include/tiny_dnn/util/util.h:24,
                 from /usr/local/include/tiny_dnn/lossfunctions/loss_function.h:12,
                 from /usr/local/include/tiny_dnn/network.h:26,
                 from /usr/local/include/tiny_dnn/tiny_dnn.h:11,
                 from /opt/sources/include/detectcone.hpp:46,
                 from /opt/sources/include/collector.hpp:29,
                 from /opt/sources/src/collector.cpp:20:
/usr/local/include/tiny_dnn/xtensor/xiterable.hpp: In member function 'xt::xiterable<D>::layout_iterator<L> xt::xiterable<D>::storage_begin()':
/usr/local/include/tiny_dnn/xtensor/xiterable.hpp:740:19: error: expected primary-expression before ')' token
   return begin<L>();
                   ^
/usr/local/include/tiny_dnn/xtensor/xiterable.hpp: In member function 'xt::xiterable<D>::layout_iterator<L> xt::xiterable<D>::storage_end()':
/usr/local/include/tiny_dnn/xtensor/xiterable.hpp:746:17: error: expected primary-expression before ')' token
   return end<L>();
                 ^
/usr/local/include/tiny_dnn/xtensor/xiterable.hpp: In member function 'xt::xiterable<D>::reverse_layout_iterator<L> xt::xiterable<D>::storage_rbegin()':
/usr/local/include/tiny_dnn/xtensor/xiterable.hpp:753:20: error: expected primary-expression before ')' token
   return rbegin<L>();
                    ^
/usr/local/include/tiny_dnn/xtensor/xiterable.hpp: In member function 'xt::xiterable<D>::reverse_layout_iterator<L> xt::xiterable<D>::storage_rend()':
/usr/local/include/tiny_dnn/xtensor/xiterable.hpp:760:18: error: expected primary-expression before ')' token
   return rend<L>();
                  ^
/opt/sources/src/detectcone.cpp: In member function 'void DetectCone::forwardDetectionORB(cv::Mat)':
/opt/sources/src/detectcone.cpp:580:38: error: array must be initialized with a brace-enclosed initializer
   cv::Mat probMap[4] = cv::Mat::zeros(m_height, m_width, CV_64F);
                        ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
make[2]: *** [CMakeFiles/cfsd18-perception-detectcone-core.dir/build.make:76: CMakeFiles/cfsd18-perception-detectcone-core.dir/src/collector.cpp.o] Error 1
cc1plus: all warnings being treated as errors
make[2]: *** [CMakeFiles/cfsd18-perception-detectcone-core.dir/build.make:63: CMakeFiles/cfsd18-perception-detectcone-core.dir/src/detectcone.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:105: CMakeFiles/cfsd18-perception-detectcone-core.dir/all] Error 2
make: *** [Makefile:141: all] Error 2
The command '/bin/sh -c mkdir -p build &&     cd build &&     cmake -D CMAKE_BUILD_TYPE=Release -D CMAKE_INSTALL_PREFIX=/tmp/cfsd18-perception-detectcone-dest .. &&     make -j3 && make test && make install && upx -9 /tmp/cfsd18-perception-detectcone-dest/bin/cfsd18-perception-detectcone' returned a non-zero code: 2
`